ui/app/mirrors/edit/[mirrorId]/syncStatusTable.tsx

```diff
@@ -53,13 +53,16 @@ const ROWS_PER_PAGE = 10;
 export const SyncStatusTable = ({ rows }: SyncStatusTableProps) => {
   const [currentPage, setCurrentPage] = useState(1);
   const totalPages = Math.ceil(rows.length / ROWS_PER_PAGE);
-
+  const [searchQuery, setSearchQuery] = useState<string>('');
   const startRow = (currentPage - 1) * ROWS_PER_PAGE;
   const endRow = startRow + ROWS_PER_PAGE;
-  const allRows = rows.slice(startRow, endRow);
-  const [displayedRows, setDisplayedRows] = useState(
-    rows.slice(startRow, endRow)
-  );
+  const displayedRows = useMemo(() => {
+    const allRows = rows.slice(startRow, endRow);
+    const shownRows = allRows.filter((row: any) => {
+      return row.batchId == parseInt(searchQuery, 10);
+    });
+    return shownRows.length > 0 ? shownRows : allRows;
```

Comment:
Can include sort here

Reply:
I've branched this from main. Once this is merged I'll rebase https://github.com/PeerDB-io/peerdb/pull/664 accordingly